### PR TITLE
Fix blockscout client and add tests

### PR DESCRIPTION
**Pull request by sevenzing**
+ Fixed blockscout client bugs and added cache to improve performance
+ Added server tests
+ Fixed timestamp conversion

### CODE REVIEW

File: blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs [hunk 1]
```diff
@@ -154,44 +154,46 @@ async fn events_from_transactions(
...
+                .remove(&TxHash::from_slice(txn.transaction_id.as_slice()))
+                .map(|t| DomainEvent {
+                    transaction_hash: t.hash,
+                    block_number: t.block,
+                    timestamp: t.timestamp,
+                    from_address: t.from.hash,
+                    method: t.method,
+                    actions: txn.actions,
+                })
+        })
```
Comment (rimrakhimov) [inline blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs:189]: In general, what this complex logic is for?

File: blockscout-ens/bens-logic/src/subgraphs_reader/blockscout.rs [hunk 2]
```diff
@@ -71,38 +80,57 @@ where
...
+        .get(
+            client
+                .url
+                .join(&format!("/api/v2/transactions/{transaction_hash:#x}"))
+                .unwrap(),
+        )
+        .send()
+        .await?;
+    Response::try_from_reqwest_response(response).await
+}
```
[the remark this answers is not in the document]
Comment (sevenzing) [reply blockscout-ens/bens-logic/src/subgraphs_reader/blockscout.rs:106]: `cached` crate doesn't support functions with `self` for some reason

File: blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs [hunk 3]
```diff
@@ -154,44 +154,46 @@ async fn events_from_transactions(
...
 ) -> Result<Vec<DomainEvent>, SubgraphReadError> {
-    let txns = txns
-        .into_iter()
-        .map(|t| (TxHash::from_slice(t.transaction_id.as_slice()), t))
-        .collect::<HashMap<_, _>>();
-    let transactions = client
-        .transactions_batch(txns.keys().collect())
+    let txn_ids: Vec<TxHash> = txns
+        .iter()
+        .map(|t| TxHash::from_slice(t.transaction_id.as_slice()))
```
[the remark this answers is not in the document]
Comment (sevenzing) [reply blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs:159]: Then I need to write Decode trait for TxHash, however I cannot impl trait for this struct, so I need to write wrapper with bunch of boilerplate... So I decided to leave it as vec<u8>

File: blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs (hunk 1 above)
Comment (sevenzing) [reply blockscout-ens/bens-logic/src/subgraphs_reader/reader.rs:189]: We only have Vec<DomainEventTransaction> struct, and need to build Vec<DomainEvent> (preserving initial order). We call transactions_batch, that will return HashMap, then for every transaction in initial vector we build DomainEvent struct

File: blockscout-ens/bens-logic/src/subgraphs_reader/blockscout.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sevenzing) [reply blockscout-ens/bens-logic/src/subgraphs_reader/blockscout.rs]: I can move this function inside BlockscoutClient::transaction function, but it looks weird

[the remark this answers is not in the document]
Comment (rimrakhimov) [reply blockscout-ens/bens-logic/src/subgraphs_reader/blockscout.rs]: Get it. No, there is no need to move it inside

**Review by rimrakhimov — APPROVED** [review]

**Pull request merged**